src/report_workflow/test_run_runner.py

```diff
@@ -0,0 +1,137 @@
+from urllib.error import HTTPError
+
+import validators
+import yaml
+
+from manifests.test_manifest import TestManifest
+from manifests.test_run_manifest import TestRunManifest
+from report_workflow.report_args import ReportArgs
+
+
+class TestRunRunner:
+    args: ReportArgs
+    test_manifest: TestManifest
+    tests_dir: str
+    test_run_manifest: TestRunManifest
+    test_run_data: dict
+
+    def __init__(self, args: ReportArgs, test_manifest: TestManifest) -> None:
+        self.args = args
+        self.base_path = args.base_path
+        self.test_manifest = test_manifest
+        self.test_run_data = self.test_run_manifest_data_template("manifest")
+        self.product_name = test_manifest.__to_dict__().get("name")
+        self.name = self.product_name.replace(" ", "-").lower()
+        self.components = self.args.components
+        self.test_run_id = args.test_run_id
+        self.test_type = self.args.test_type
+        self.test_manifest_path = self.args.test_manifest_path
+        self.artifact_paths = ""
+        for k, v in self.args.artifact_paths.items():
+            self.artifact_paths = " ".join([self.artifact_paths, k + "=" + v]).strip(" ")
+
+        self.dist_manifest = "/".join([self.args.artifact_paths[self.name], "dist", self.name, "manifest.yml"]) if self.args.artifact_paths[self.name].startswith("https://") \
+            else os.path.join(self.args.artifact_paths[self.name], "dist", self.name, "manifest.yml")
+        self.test_components = self.test_manifest.components
+
+    def update_data(self) -> dict:
+        self.test_run_data["name"] = self.product_name
+        self.test_run_data["test-run"] = self.update_test_run_data()
+        for component in self.test_components.select(focus=self.args.components):
+            self.test_run_data["components"].append(self.component_entry(component.name))
+        return self.test_run_data
+
+    def update_test_run_data(self) -> dict:
+        test_run_data = {
+            "Command": generate_test_command(self.test_type, self.test_manifest_path, self.artifact_paths),
+            "TestType": self.test_type,
+            "TestManifest": self.test_manifest_path,
+            "DistributionManifest": self.dist_manifest,
+            "TestID": str(self.test_run_id)
+        }
+        return test_run_data
+
+    def generate_report(self, data: dict, output_dir: str) -> Any:
+        test_run_manifest = TestRunManifest(data)
+        test_run_manifetest_run_manifest_file = os.path.join(output_dir, "test-run.yml")
+        logging.info(f"Generating test-run.yml in {output_dir}")
+        return test_run_manifest.to_file(test_run_manifetest_run_manifest_file)
+
+    def component_entry(self, component_name: str) -> Any:
+        component = self.test_run_manifest_data_template("component")
+        component["name"] = component_name
+        component["command"] = generate_test_command(self.test_type, self.test_manifest_path, self.artifact_paths, component_name)
```

Comment:
you already have test command added to `self.test_run_data["test-run"]`, why not use from there?

Reply:
I'm generating the test command dynamically each time. Basically we are adding one optional argument `component name` here so it would generate the command with one more argument. I didn't save the command from the `test-run` entry so each time we are running `generate_test_command` to create it.